plonk-core/src/proof_system/proof.rs

```diff
@@ -108,10 +113,10 @@ where
     PC: HomomorphicCommitment<F>,
 {
     /// Performs the verification of a [`Proof`] returning a boolean result.
-    pub(crate) fn verify<P>(
+    pub(crate) fn verify<P, D: Digest>(
         &self,
         plonk_verifier_key: &PlonkVerifierKey<F, PC>,
-        transcript: &mut Transcript,
+        _transcript: &mut Transcript,
```

Comment:
It does not make any sense to pass the `Transcript` anymore.

This is the part that I don't really like TBH. Getting rid of the Transcript from `merlin` and just using the FiatShamir internally prevents the parties to setup some original randomness on the transcript before starting the protocol.

On the other hand, we get rid off the labes which we saw was a bit annoying to workarround in #133 .
Also, it's much easier for what refers to the next refactor we have planned where we want to support arbitrary ammount of wires.

Reply:
Sorry for the miscommunication, I spoke with @davidnevadoc and @LukePearson1. I intentionally didn't remove this transcript, as you said change is already really big and I wanted to check if that makes sense at all before introducing even more changes. 

Instead, rng will be passed as argument and it will be also preprocessed with same values that are being inserted in transcript now. 

Happy to chat more about arbitrary number of wires.